avm/res/compute/virtual-machine-scale-set/main.bicep

```diff
@@ -533,6 +595,28 @@ resource vmss 'Microsoft.Compute/virtualMachineScaleSets@2023-09-01' = {
             : null
         }
       }
+      extensionProfile: extensionHealthConfig.enabled
+        ? {
+            extensions: [
+              {
+                name: 'HealthExtension'
+                properties: {
+                  publisher: 'Microsoft.ManagedServices'
+                  type: (osType == 'Windows' ? 'ApplicationHealthWindows' : 'ApplicationHealthLinux')
+                  typeHandlerVersion: '1.0'
+                  autoUpgradeMinorVersion: false
+                  settings: {
+                    protocol: contains(extensionHealthConfig, 'protocol') ? extensionHealthConfig.protocol : 'http'
+                    port: contains(extensionHealthConfig, 'port') ? extensionHealthConfig.port : '80'
+                    requestPath: contains(extensionHealthConfig, 'requestPath')
+                      ? extensionHealthConfig.requestPath
+                      : '/'
```

Comment:
@AlexanderSehr if I deploy through child module, it will be deployed after the VMSS. In order for some features to work, the health extension must be deployed with the VMSS deployment. Implemented all the suggested changes. If you enable the healthextension via Portal, the autoUpgradeMinorVersion is set to false, so I mimiced that one. Updated it to your suggestion though.